### ISSUE TITLE
We get with the last release some errors. Datastore is not yet connected

### ISSUE THREAD

**Issue by Trackhe** [baileymh/pf1e-content#16]
We using this docker image: https://github.com/felddy/foundryvtt-docker
and the files are Present.

Debug.log:
```
{"level":"error","timestamp":"2021-06-17 20:23:50","message":"The pf-content.pf-35-content Datastore is not yet connected!","stack":"Error: The pf-content.pf-35-content Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at Function.find (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:1375)\n    at ServerDatabaseBac>
{"level":"error","timestamp":"2021-06-17 20:23:50","message":"The pf-content.pf-35-content Datastore is not yet connected!","stack":"Error: The pf-content.pf-35-content Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:975)\n    >
{"level":"error","timestamp":"2021-06-17 20:23:50","message":"The pf-content.pf-artifacts Datastore is not yet connected!","stack":"Error: The pf-content.pf-artifacts Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:975)\n    at>
{"level":"error","timestamp":"2021-06-17 20:23:50","message":"The pf-content.pf-buffs Datastore is not yet connected!","stack":"Error: The pf-content.pf-buffs Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:975)\n    at ServerD>
{"level":"error","timestamp":"2021-06-17 20:23:50","message":"The pf-content.pf-traps-and-haunts Datastore is not yet connected!","stack":"Error: The pf-content.pf-traps-and-haunts Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:>
{"message":"FoundryVTT | Deleted 0 Drawing documents with ids: [] in parent Scene [**********","level":"info","timestamp":"2021-06-17 20:23:51"}
{"level":"error","timestamp":"2021-06-17 20:24:05","message":"The pf-content.pf-35-content Datastore is not yet connected!","stack":"Error: The pf-content.pf-35-content Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at Function.find (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:1375)\n    at ServerDatabaseBac>
{"level":"error","timestamp":"2021-06-17 20:24:06","message":"The pf-content.pf-35-content Datastore is not yet connected!","stack":"Error: The pf-content.pf-35-content Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at Function.find (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:1375)\n    at ServerDatabaseBac>
{"message":"Vended World data to User [**********] in 18ms","level":"info","timestamp":"2021-06-17 20:25:22"}
{"level":"error","timestamp":"2021-06-17 20:25:23","message":"The pf-content.pf-35-content Datastore is not yet connected!","stack":"Error: The pf-content.pf-35-content Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:975)\n    >
{"level":"error","timestamp":"2021-06-17 20:25:23","message":"The pf-content.pf-artifacts Datastore is not yet connected!","stack":"Error: The pf-content.pf-artifacts Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:975)\n    at>
{"level":"error","timestamp":"2021-06-17 20:25:23","message":"The pf-content.pf-buffs Datastore is not yet connected!","stack":"Error: The pf-content.pf-buffs Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:975)\n    at ServerD>
{"level":"error","timestamp":"2021-06-17 20:25:23","message":"The pf-content.pf-traps-and-haunts Datastore is not yet connected!","stack":"Error: The pf-content.pf-traps-and-haunts Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:>
{"level":"error","timestamp":"2021-06-17 20:25:57","message":"The pf-content.pf-35-content Datastore is not yet connected!","stack":"Error: The pf-content.pf-35-content Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at Function.find (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:1375)\n    at ServerDatabaseBac>
{"level":"error","timestamp":"2021-06-17 20:25:58","message":"The pf-content.pf-35-content Datastore is not yet connected!","stack":"Error: The pf-content.pf-35-content Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at Function.find (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:1375)\n    at ServerDatabaseBac>
{"message":"Created client session **********","level":"info","timestamp":"2021-06-17 21:00:42"}
{"session":"**********","ip":"::ffff:172.18.0.1","level":"info","message":"User authentication successful for user Lucas","timestamp":"2021-06-17 21:03:50"}
{"message":"Vended World data to User [**********] in 14ms","level":"info","timestamp":"2021-06-17 21:03:51"}
{"level":"error","timestamp":"2021-06-17 21:03:52","message":"The pf-content.pf-35-content Datastore is not yet connected!","stack":"Error: The pf-content.pf-35-content Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:975)\n    >
{"level":"error","timestamp":"2021-06-17 21:03:52","message":"The pf-content.pf-artifacts Datastore is not yet connected!","stack":"Error: The pf-content.pf-artifacts Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:975)\n    at>
{"level":"error","timestamp":"2021-06-17 21:03:52","message":"The pf-content.pf-buffs Datastore is not yet connected!","stack":"Error: The pf-content.pf-buffs Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:975)\n    at ServerD>
{"level":"error","timestamp":"2021-06-17 21:03:52","message":"The pf-content.pf-traps-and-haunts Datastore is not yet connected!","stack":"Error: The pf-content.pf-traps-and-haunts Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:>
{"message":"Vended World data to User [**********] in 11ms","level":"info","timestamp":"2021-06-17 21:04:13"}
{"message":"Vended World data to User [**********] in 14ms","level":"info","timestamp":"2021-06-17 21:04:14"}
```

Error.log:
```
...
{"level":"error","timestamp":"2021-06-17 20:16:16","message":"ENOENT: no such file or directory, rename '/data/Data/modules/pf-content/packs/pf-class-abilities.db~' -> '/data/Data/modules/pf-content/packs/pf-class-abilities.db'","stack":"Error: ENOENT: no such file or directory, rename '/data/Data/modules/pf-content/packs/pf-class-abilities.db~' -> '/data/Data/modules/pf-content/packs/pf-class-abilities.db'"}
{"level":"error","timestamp":"2021-06-17 20:16:16","message":"ENOENT: no such file or directory, rename '/data/Data/modules/pf-content/packs/pf-class-abilities.db~' -> '/data/Data/modules/pf-content/packs/pf-class-abilities.db'","stack":"Error: ENOENT: no such file or directory, rename '/data/Data/modules/pf-content/packs/pf-class-abilities.db~' -> '/data/Data/modules/pf-content/packs/pf-class-abilities.db'"}
{"level":"error","timestamp":"2021-06-17 20:16:17","message":"ENOENT: no such file or directory, rename '/data/Data/modules/pf1e-archetypes/packs/pf-arch-features.db~' -> '/data/Data/modules/pf1e-archetypes/packs/pf-arch-features.db'","stack":"Error: ENOENT: no such file or directory, rename '/data/Data/modules/pf1e-archetypes/packs/pf-arch-features.db~' -> '/data/Data/modules/pf1e-archetypes/packs/pf-arch-features.db'"}
{"level":"error","timestamp":"2021-06-17 20:16:17","message":"ENOENT: no such file or directory, rename '/data/Data/modules/pf1e-archetypes/packs/pf-arch-features.db~' -> '/data/Data/modules/pf1e-archetypes/packs/pf-arch-features.db'","stack":"Error: ENOENT: no such file or directory, rename '/data/Data/modules/pf1e-archetypes/packs/pf-arch-features.db~' -> '/data/Data/modules/pf1e-archetypes/packs/pf-arch-features.db'"}
{"level":"error","timestamp":"2021-06-17 20:16:17","message":"ENOENT: no such file or directory, rename '/data/Data/modules/pf-content/packs/pf-companions.db~' -> '/data/Data/modules/pf-content/packs/pf-companions.db'","stack":"Error: ENOENT: no such file or directory, rename '/data/Data/modules/pf-content/packs/pf-companions.db~' -> '/data/Data/modules/pf-content/packs/pf-companions.db'"}
{"level":"error","timestamp":"2021-06-17 20:16:17","message":"ENOENT: no such file or directory, rename '/data/Data/modules/pf-content/packs/pf-companions.db~' -> '/data/Data/modules/pf-content/packs/pf-companions.db'","stack":"Error: ENOENT: no such file or directory, rename '/data/Data/modules/pf-content/packs/pf-companions.db~' -> '/data/Data/modules/pf-content/packs/pf-companions.db'"}
{"level":"error","timestamp":"2021-06-17 20:22:42","message":"The pf-content.pf-35-content Datastore is not yet connected!","stack":"Error: The pf-content.pf-35-content Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:975)\n    >
{"level":"error","timestamp":"2021-06-17 20:22:42","message":"The pf-content.pf-artifacts Datastore is not yet connected!","stack":"Error: The pf-content.pf-artifacts Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:975)\n    at>
{"level":"error","timestamp":"2021-06-17 20:22:42","message":"The pf-content.pf-buffs Datastore is not yet connected!","stack":"Error: The pf-content.pf-buffs Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:975)\n    at ServerD>
{"level":"error","timestamp":"2021-06-17 20:22:42","message":"The pf-content.pf-traps-and-haunts Datastore is not yet connected!","stack":"Error: The pf-content.pf-traps-and-haunts Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:>
{"level":"error","timestamp":"2021-06-17 20:22:49","message":"The pf-content.pf-35-content Datastore is not yet connected!","stack":"Error: The pf-content.pf-35-content Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:975)\n    >
{"level":"error","timestamp":"2021-06-17 20:22:49","message":"The pf-content.pf-artifacts Datastore is not yet connected!","stack":"Error: The pf-content.pf-artifacts Datastore is not yet connected!\n    at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)\n    at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:975)\n    at>
...
```

**Comment by baileymh** [comment]
Hey Trackhe, 

Unfortunately, I'm not sure why this error is occurring. I'm still running on .7.10, and haven't migrated to the .8.x versions yet. I saw a comment in the discord about someone experiencing this issue (not sure if that was you). 

The only things I can think of are a problem with read/write permissions for the folder, or that there is a mod conflict. Can you try running with all modules besides pf-content turned off and see if the error persists?

**Comment by Trackhe** [comment]
I cant srry. I set the server up for a friend. After some investigating,  he say he clicked on something update in Foundry. after that it is going working. I dont know what exactly srry.

**Comment by Teifun2** [comment]
Hi

We seem to run into a smilar issue.  

1. We also use the same docker image https://github.com/felddy/foundryvtt-docker
2. foundry version: 0.8.9
3. pf1-content version: 0.2.2

As soon as i activate pf1-content we get the following error in the server logs:

```
FoundryVTT | 2021-11-24 12:01:25 | [[31merror[39m] The pf-content.pf-companions Datastore is not yet connected!
Error: The pf-content.pf-companions Datastore is not yet connected!

at Function.get db [as db] (file:///home/foundry/resources/app/dist/database/backend/server-document.mjs:1:625)
at ServerDatabaseBackend._getDocuments (file:///home/foundry/resources/app/dist/database/backend/server-backend.mjs:1:1032)
at Function.getIndex (file:///home/foundry/resources/app/dist/database/backend/server-compendium.mjs:1:1139)
at Function.vend (file:///home/foundry/resources/app/dist/packages/world.mjs:1:9179)
at runMicrotasks (<anonymous>)
at runNextTicks (internal/process/task_queues.js:60:5)
at processImmediate (internal/timers.js:437:9)
at async Function.requestWorldData (file:///home/foundry/resources/app/dist/packages/world.mjs:1:7519)
FoundryVTT | 2021-11-24 12:01:37 | [[31merror[39m] The pf-content.pf-companions Datastore is not yet connected!
```

Any idea what this could mean?

**Comment by Trackhe** [comment]
I guess you should open a own ticket. 

My fix was waiting. My user had fix this with an option on the Webinterface. I dont really know how but i changed nothing in the container because the files exists for me and the rights are right too.

Take care of that and try out. you cant destroy anything ^^

**Comment by Teifun2** [comment]
@Trackhe Its wierd. I downgraded the foundry version and upgraded again and suddenly it worked. So it is also solved for me but i have no idea how this could be reproduced. I reinstalled the module multiple times before.

**Comment by chris-sanders** [comment]
Just stopping by b/c I got this error running version 9.242. I removed and reinstalled the module based on the help message to be sure nothing was wrong. The thing is I can't remember for sure but I think it was only after doing the clean install that I got the failure to load the database. Reverting to 0.8.9 and then back to 9.242 and everything loads fine.

Maybe there's a schema conversion that happens if foundry boots in version 9 after having been running in version 8? It seems to me that installing the modules on 9.x directly always fails but I can't find any particularly useful logging to dig any deeper than I have. I love all of the content you have, but right now the updater doesn't work and I'm not sure if there's a way I can contribute back running the current stable version.
